src/SimpleCacheTest.php

```diff
@@ -0,0 +1,678 @@
+                $this->assertNull($r);
+            } else {
+                $this->assertFalse(true, 'This should not happend');
+            }
+        }
+        $this->assertEquals('value0', $this->cache->get('key0'));
+        $this->assertNull($this->cache->get('key1'));
+    }
+
+    public function testDeleteMultiple()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+        }
+
+        $this->assertTrue($this->cache->deleteMultiple([]), 'Deleting a empty array should return true');
+        $this->assertTrue($this->cache->deleteMultiple(['key']), 'Deleting a value that does not exist should return true');
+
+        $this->cache->set('key0', 'value0');
+        $this->cache->set('key1', 'value1');
+        $this->assertTrue($this->cache->deleteMultiple(['key0', 'key1']), 'Delete must return true on success');
+        $this->assertNull($this->cache->get('key0'), 'Values must be deleted on deleteMultiple()');
+        $this->assertNull($this->cache->get('key1'), 'Values must be deleted on deleteMultiple()');
+    }
+
+    public function testDeleteMultipleGenerator()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+        }
+
+        $gen = function () {
+            yield 'key0';
+            yield 'key1';
+        };
+        $this->cache->set('key0', 'value0');
+        $this->assertTrue($this->cache->deleteMultiple($gen()), 'Deleting a generator should return true');
+
+        $this->assertNull($this->cache->get('key0'), 'Values must be deleted on deleteMultiple()');
+        $this->assertNull($this->cache->get('key1'), 'Values must be deleted on deleteMultiple()');
+    }
+
+    public function testHas()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+        }
+
+        $this->assertFalse($this->cache->has('key0'));
+        $this->cache->set('key0', 'value0');
+        $this->assertTrue($this->cache->has('key0'));
+    }
+
+    /**
+     * @expectedException \Psr\SimpleCache\InvalidArgumentException
+     * @dataProvider invalidKeys
+     */
+    public function testGetInvalidKeys($key)
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+        }
+
+        $this->cache->get($key);
+    }
+
+    /**
+     * @expectedException \Psr\SimpleCache\InvalidArgumentException
+     * @dataProvider invalidKeys
+     */
+    public function testGetMultipleInvalidKeys($key)
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+        }
+
+        $result = $this->cache->getMultiple(['key1', $key, 'key2']);
+        foreach ($result as $r) {
+            // We want to make sure we iterate over the results
```

Comment:
why are we iterating here ? The spec says that ``getMultiple`` itself is throwing. Your test allows throwing during the iteration instead, which would not be catched by a try/catch around the call to ``getMultiple`` (and yes, we took care of this in Symfony as the first implementation of our PSR-6 cache was precisely suffering of this issue)